src/lib/components/SvgSnakeBody.svelte

```diff
@@ -26,153 +33,218 @@
       body.pop();
     }
 
+    // Split wrapped body parts into separated segments
+    const bodySegments = ((body: Point[]): Point[][] => {
+      if (body.length == 0) {
+        return [[]];
+      }
+
+      let prev = body[0];
+      const segments: Point[][] = [[prev]];
+
+      for (let i = 1; i < body.length; i++) {
+        const cur = body[i];
+
+        // Start new segment
+        if (!isAdjacentPoint(cur, prev)) {
+          segments.push([]);
+        }
+
+        segments[segments.length - 1].push(cur);
+        prev = cur;
+      }
+      return segments;
+    })(body);
+
+    const enrichSvgCellCenter = (p: Point): SvgPointWithCircleProps => {
+      const c = svgCalcCellCenter(svgCalcParams, p);
+      return {
+        cx: c.x,
+        cy: c.y,
+        ...p
+      };
+    };
+
     // Get the center point of each body square we're going to render
-    const bodyCenterPoints = body.map((p) => {
-      const svgCenter = svgCalcCellCenter(svgCalcParams, p);
-      return { cx: svgCenter.x, cy: svgCenter.y, x: p.x, y: p.y };
-    });
+    const bodySegmentsCenterPoints = bodySegments.map((segment) =>
+      segment.map(enrichSvgCellCenter)
+    );
 
     // If we're drawing *any* body, we want to extend the first and last points
     // to connect to the head and tail across the cell spacing.
-    if (bodyCenterPoints.length > 0) {
-      // Use overlap to ensure that we connect to head and tail.
+    if (body.length > 0) {
       const gapSize = svgCalcParams.cellSpacing + OVERLAP;
 
+      // Extend each wrapped segment endings towards border
+      for (let i = 0; i < bodySegmentsCenterPoints.length; i++) {
+        // Extend each segment last point towards border
+        if (i < bodySegmentsCenterPoints.length - 1) {
+          const cur = bodySegmentsCenterPoints[i].slice(-1)[0];
```

Comment:
NitPick: We can probably use `at(-1)` instead of the `slice(-1)[0]` but 🤷

Reply:
As we discussed it online refactored obtaining last items with util function `last(arr)`